container/Dockerfile

```diff
@@ -1,30 +1,22 @@
 FROM alpine:latest
-EXPOSE 53 53/udp
 
-RUN apk --update upgrade && apk add bind
+EXPOSE 53 53/udp
+COPY entrypoint.sh /
+ENTRYPOINT ["/entrypoint.sh"]
 
-# /etc/bind needs to be owned by root, group owned by "bind", and chmod 750
-# since we are mounting, do it manually
-# NOTE: Per Dockerfile manual --> need to mkdir the mounted dir to chown
-# &
-# /var/bind needs to be owned by root, group owned by "bind", and chmod 770
-# since we are mounting, do it manually
-# NOTE: Per Dockerfile manual --> need to mkdir the mounted dir to chown
-# &
-# Get latest bind.keys
-RUN mkdir -m 0770 -p /etc/bind && chown -R root:named /etc/bind ; \
-    mkdir -m 0770 -p /var/bind && chown -R root:named /var/cache ; \
-    wget -q -O /etc/bind/bind.keys https://ftp.isc.org/isc/bind9/keys/9.11/bind.keys.v9_11 ; \
-    rndc-confgen -a -r /dev/urandom
+RUN apk --update upgrade && \
+    apk add bind gnupg1 && \
+    rm -rf /var/cache/apk/* && \
+    chmod g-w /var/bind && \
```

Comment:
The default alpine bind install sets up:
```
drwxrwx---    5 root     named       4.0K Feb 16 02:19 bind
```

Any reason you are removing group writes?

Also - they do not setup a ```/var/cache/bind``` at all in alpine.

So the question here really is do we follow "standards" (and again, I am using this loosely, since there don't seem be any official ones, but mostly popular opinion/common deployments), or do we go with what alpine sets up by default?

Reply:
I'm removing the group write because I don't want anything running as named writing to this directory.

I don't know why you are so fixated on using `/var/bind` at all. Is it just because whoever created the `alpline` package moved the files there? All we need is a directory that `named` can write to so that's what I created.

Reply:
If you are using dynamic zones, you need to have the group with write permissions if that's where the zones will live.

I am not fixated on using /var/bind -- in fact, it's very much going against what feels correct to me. Every production service we run (and have run for the past 10+ years) is using /var/cache/bind, but again for better or worse, this is what the OS/maintainers have chosen as their standard. Personally - I would like to stick something that's generally accepted among deployments, or at least within the OS used.

Imo - that's /etc/bind for all configs, and /var/cache/bind for all zones. I've been told that /var/cache/bind is not used outside of Debian/Ubuntu, and that '/var/bind' is used as a default (RHEL, CentOS, Alpine, etc). OpenBSD still uses /var/named.

So for example, one thing that everyone agrees on is that the run pid should be:
```/var/run/named/named.pid```

see for ref:

(ubuntu and debian -> /var/cache/bind)
https://wiki.debian.org/Bind9
https://help.ubuntu.com/community/BIND9ServerHowto

vs

(rhel/centos/openbsd-> /var/named)
https://man.openbsd.org/OpenBSD-5.5/named.8
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/deployment_guide/s2-bind-zone
(and in this case, they even explicitly mention bind is now allowed to write to it -- so no dynamic zones in the same dir)

Reply:
I fixed the PID path already. We could really be odd and just use `/var/bind/cache` if you want.

Reply:
:) 

Give me some time to go through everything (in the middle of leaving for a vacation). There are just a couple of things I think we should change, but the rest looks good. I really like some of the suggestions. Thanks for the time/input/work and for going through everything!